lib/views/request/request_view.dart

```diff
@@ -18,13 +22,26 @@ class RequestView extends StatefulWidget {
 class _RequestViewState extends State<RequestView> {
   String display = '0';
 
-  Future<AppGenerateInvoice> generateInvoice() async {
+  Future<AppGenerateInvoice?> generateInvoice() async {
     String label = '${DateTime.now()}';
     String description =
-        "This is a new description created at${DateTime.now()}";
-    AppGenerateInvoice response = await widget.provider
-        .get<AppApi>()
-        .generateInvoice(label, description, amount: int.parse(display));
+        "This is a new description created at ${DateTime.now()}";
+    AppGenerateInvoice response;
+    try {
+      response = await widget.provider
+          .get<AppApi>()
+          .generateInvoice(label, description, amount: int.parse(display));
+    } catch (e) {
+      var jsonString = e.toString().substring(e.toString().indexOf('{'));
+      ErrorDecoder decoder = ErrorDecoder.fromJSON(jsonDecode(jsonString));
+      GlobalComponent.showAppDialog(
+          context: context,
+          title: 'Invalid rune',
+          message: decoder.message,
+          closeMsg: 'Ok',
+          imageProvided: const AssetImage('assets/images/exclamation.png'));
```

Comment:
wondering if this can be injected inside a function and avoid to use the same code everywhere?

Reply:
Sure